libs/ledger-live-common/src/device/use-cases/appDataBackup/restoreAppData.ts

```diff
@@ -51,14 +64,24 @@ export function restoreAppData(
           // Commit the restore process, last step
           await restoreAppStorageCommit(transport);
 
-          // NOTE: DELETE DATA
-
           subscriber.next({
             type: RestoreAppDataEventType.AppDataRestored,
           });
-          subscriber.complete();
+        }),
+        // Delete the app data from the storage
+        switchMap(() => deleteAppData(appName, deviceModelId, storageProvider)),
```

Comment:
Interaction with data storage in this file is counter to the initial pattern.
This `restoreAppData` as well as `backupAppData` are dedicated to data stream programming. I think that the data deletion use case should be enchained in `restoreAppDataUseCase` where all the parameters are already present.

Reply:
In addition, the deleting the local data is not part of the data restoration workflow. It is an operation independent and should not be injected into normal restoration workflow.

Reply:
make sense, i will fix this